pkg/service/roomservice.go

```diff
@@ -232,10 +231,20 @@ func (s *RoomService) MutePublishedTrack(ctx context.Context, req *livekit.MuteR
 
 func (s *RoomService) UpdateParticipant(ctx context.Context, req *livekit.UpdateParticipantRequest) (*livekit.ParticipantInfo, error) {
 	AppendLogFields(ctx, "room", req.Room, "participant", req.Identity)
-	maxMetadataSize := int(s.roomConf.MaxMetadataSize)
+	maxMetadataSize := int(s.limitConf.MaxMetadataSize)
 	if maxMetadataSize > 0 && len(req.Metadata) > maxMetadataSize {
 		return nil, twirp.InvalidArgumentError(ErrMetadataExceedsLimits.Error(), strconv.Itoa(maxMetadataSize))
 	}
+	maxAttributeSize := int(s.limitConf.MaxAttributesSize)
+	if maxAttributeSize > 0 {
+		total := 0
+		for key, val := range req.Attributes {
+			total += len(key) + len(val)
+		}
+		if total > maxAttributeSize {
+			return nil, twirp.InvalidArgumentError(ErrAttributeExceedsLimits.Error(), strconv.Itoa(maxAttributeSize))
+		}
+	}
```

Comment:
we can't know the total attributes size until we've merged the updates with the current values. do we want to duplicate the validation here knowing that this is unreliable?

Reply:
just trying to save the serialization & trip across RPCs if it's already large. we do check again on the media size